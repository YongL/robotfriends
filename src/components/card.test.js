import { shallow } from 'enzyme';
// import { shallow, mount, render, configure } from 'enzyme';
import React from 'react'; // import this because it is jsx
import Card from './card';


// import Adapter from 'enzyme-adapter-react-16';

// configure({adapter: new Adapter()});

// the test script come with react script /// create-react-app 


// shallow only shollow or render one give component it doesn't care anything inside or children
// inside it
// the document doesn't really give a specific direction of where to 
// to put the configure// 
// under src create setupTests.js 
// https://airbnb.io/enzyme/docs/installation/index.html


// it('render Card', () => expect(shallow(<Card />).length).toBe(1));


it('render Card', () => expect(shallow(<Card />)).toMatchSnapshot());

// jest do snaphot testing, which snapshot the codes (string copy not image)

