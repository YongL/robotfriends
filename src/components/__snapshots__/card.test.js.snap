// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Card 1`] = `
<div
  className="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
>
  <img
    alt="robots"
    src="https://robohash.org/undefined?size=200x200"
  />
  <div />
  <div />
  <div>
    more stuff coming
  </div>
</div>
`;
